diff -rwu gnu.hylafax-0.0.8-org/src/gnu/hylafax/HylaFAXClientProtocol.java gnu.hylafax-0.0.8/src/gnu/hylafax/HylaFAXClientProtocol.java
--- gnu.hylafax-0.0.8-org/src/gnu/hylafax/HylaFAXClientProtocol.java	2003-12-29 20:44:10.000000000 +0100
+++ gnu.hylafax-0.0.8/src/gnu/hylafax/HylaFAXClientProtocol.java	2005-09-30 04:43:36.000000000 +0200
@@ -851,11 +851,19 @@
       ostream.flush();
       response= istream.readLine();
       st= new StringTokenizer(response);
-      if(!st.nextToken().equals("213")){
+      if (response.startsWith("213")) {
+    	  // FIXME check response.length()
+          String res = response.substring(4);
+          while (response.charAt(3) == '-') {
+              response= istream.readLine();
+              if (!response.startsWith("213"))
          throw (new ServerResponseException(response));
+        	  // FIXME check response.length()
+              res += "\n" + response.substring(4);
       }
-
-      return st.nextToken();
+          return res;
+      } else
+          throw (new ServerResponseException(response));
    }// jparm
 
 
diff -rwu gnu.hylafax-0.0.8-org/src/gnu/inet/ftp/Putter.java gnu.hylafax-0.0.8/src/gnu/inet/ftp/Putter.java
--- gnu.hylafax-0.0.8-org/src/gnu/inet/ftp/Putter.java	2002-09-23 21:35:08.000000000 +0200
+++ gnu.hylafax-0.0.8/src/gnu/inet/ftp/Putter.java	2005-04-19 23:54:48.000000000 +0200
@@ -179,7 +179,7 @@
     * @param event the event to distribute to each ConnectionListener
     **/
    protected void signalConnectionOpened(ConnectionEvent event){
-      Enumeration listeners= transferListeners.elements();
+      Enumeration listeners= connectionListeners.elements();
       while(listeners.hasMoreElements()){
          ConnectionListener listener=
             (ConnectionListener)listeners.nextElement();
@@ -192,7 +192,7 @@
     * @param event the event to distribute to each ConnectionListener
     **/
    protected void signalConnectionClosed(ConnectionEvent event){
-      Enumeration listeners= transferListeners.elements();
+      Enumeration listeners= connectionListeners.elements();
       while(listeners.hasMoreElements()){
          ConnectionListener listener=
             (ConnectionListener)listeners.nextElement();
@@ -205,7 +205,7 @@
     * @param exception the exception that was thrown
     **/
    protected void signalConnectionFailed(Exception exception){
-      Enumeration listeners= transferListeners.elements();
+      Enumeration listeners= connectionListeners.elements();
       while(listeners.hasMoreElements()){
          ConnectionListener listener=
             (ConnectionListener)listeners.nextElement();
